src/ILCompiler.TypeSystem/tests/TestMetadataFieldLayoutAlgorithm.cs

```diff
@@ -31,5 +31,10 @@ protected override void FinalizeRuntimeSpecificStaticFieldLayout(TypeSystemConte
                 layout.ThreadStatics.Size = LayoutInt.Zero;
             }
         }
+
+        protected override ComputedInstanceFieldLayout ComputeAutoFieldLayout(MetadataType type, int numInstanceFields)
+        {
+            return ComputeSequentialFieldLayout(type, numInstanceFields);
```

Comment:
How are we testing auto-layout in the type system tests if we enforce sequential layout here? If we let tests use auto layout here and force the existing sequential tests to have sequential layout (which it looks like you did) we should be able to test both

Reply:
As long as using the AutoLayout algorithm in this class doesn't adversely affect the mainline compiler, then I can remove this override so we can actually run the tests in our managed UnitTests.

Reply:
I'll go ahead and remove this override so that the TestMetadataFieldLayoutAlgorithm uses the actual AutoLayout algorithm.

Reply:
The mainline CoreRT compiler uses sequential layout, yes. The type system tests should be more agnostic and cover all uses of the type system.